pebblo/app/service/discovery/discovery_service.py

```diff
@@ -0,0 +1,259 @@
+        if load_id:
+            AppClass = AiDataLoaderTable
+            app_type = ApplicationTypes.LOADER.value
+        else:
+            AppClass = AiAppTable
+            app_type = ApplicationTypes.RETRIEVAL.value
+
+        return app_type, AppClass
+
+    def _fetch_chain_details(self, app_metadata) -> list[Chain]:
+        """
+        Retrieve chain details from input data and return its corresponding model object.
+        """
+        # TODO: Discussion on the uniqueness of chains is not done yet,
+        #  so for now we are appending chain to existing chains in the file for this app.
+        logger.debug("Updating app chains details from input chain details")
+        chains = list()
+
+        if app_metadata:
+            chains = app_metadata.get("chains", [])
+
+        for chain in self.data.get("chains", []):
+            name = chain["name"]
+            model = chain["model"]
+            # vector db details
+            vector_db_details = []
+            for vector_db in chain.get("vector_dbs", []):
+                vector_db_obj = VectorDB(
+                    name=vector_db.get("name"),
+                    version=vector_db.get("version"),
+                    location=vector_db.get("location"),
+                    embeddingModel=vector_db.get("embedding_model"),
+                    pkgInfo=None,
+                )
+
+                package_info = vector_db.get("pkg_info")
+                if package_info:
+                    pkg_info_obj = PackageInfo(
+                        projectHomePage=package_info.get("project_home_page"),
+                        documentationUrl=package_info.get("documentation_url"),
+                        pypiUrl=package_info.get("pypi_url"),
+                        licenceType=package_info.get("licence_type"),
+                        installedVia=package_info.get("installed_via"),
+                        location=package_info.get("location"),
+                    )
+                    vector_db_obj.pkgInfo = pkg_info_obj
+
+                vector_db_details.append(vector_db_obj)
+            chain_obj = Chain(name=name, model=model, vectorDbs=vector_db_details)
+            chains.append(chain_obj.dict())
+
+        logger.debug(f"Application Name [{self.app_name}]")
+        return chains
+
+    def _fetch_retrievals_details(self, app_metadata) -> list:
+        """
+        Retrieve existing retrievals details from metadata file and append the new retrieval details
+        """
+        logger.debug("Updating app retrievals details with input retrieval details")
+        retrievals_details = list()
+
+        if app_metadata:
+            retrievals_details = app_metadata.get("retrievals", [])
+
+        input_retrievals_details = self.data.get("retrievals", [])
+        retrievals_details.extend(input_retrievals_details)
+
+        return retrievals_details
+
+    @timeit
+    def process_request(self, data):
+        try:
+            self.db = SQLiteClient()
```

Comment:
We are creating SqliteClient in every request. Is it possible to create a db client once and use it across? Maybe a Singleton class or any other option?